src/objectives/elbo/scoregradelbo.jl

```diff
@@ -0,0 +1,138 @@
+  \\nabla_{\\lambda} \\log q_{\\lambda}(z) \\left(\\pi\\left(z\\right) - \\beta\\right)
+\\right]
+```
+
+# Arguments
+- `n_samples::Int`: Number of Monte Carlo samples used to estimate the ELBO.
+
+# Keyword Arguments
+- `entropy`: The estimator for the entropy term. (Type `<: AbstractEntropyEstimator`; Default: `ClosedFormEntropy()`)
+- `baseline_window_size::Int`: The window size to use to compute the baseline. (Default: `10`)
+- `baseline_history::Vector{Float64}`: The history of the baseline. (Default: `Float64[]`)
+
+# Requirements
+- The variational approximation ``q_{\\lambda}`` implements `rand` and `logpdf`.
+- `logpdf(q, x)` must be differentiable with respect to `q` by the selected AD backend.
+- The target distribution and the variational approximation have the same support.
+
+Depending on the options, additional requirements on ``q_{\\lambda}`` may apply.
+"""
+struct ScoreGradELBO{EntropyEst <: AbstractEntropyEstimator} <:
+	   AdvancedVI.AbstractVariationalObjective
+	entropy::EntropyEst
+	n_samples::Int
+    baseline_window_size::Int
+    baseline_history::Vector{Float64}
+end
+
+function ScoreGradELBO(
+	n_samples::Int;
+	entropy::AbstractEntropyEstimator = ClosedFormEntropy(),
+    baseline_window_size::Int = 10,
+    baseline_history::Vector{Float64} = Float64[]
+)
+	ScoreGradELBO(entropy, n_samples, baseline_window_size, baseline_history)
+end
+
+
+function Base.show(io::IO, obj::ScoreGradELBO)
+	print(io, "ScoreGradELBO(entropy=")
+	print(io, obj.entropy)
+	print(io, ", n_samples=")
+	print(io, obj.n_samples)
+    print(io, ", baseline_window_size=")
+    print(io, obj.baseline_window_size)
+	print(io, ")")
+end
+
+function compute_control_variate_baseline(history, window_size)
+    if length(history) == 0
+        return 1.0
+    end
+    min_index = max(1, length(history) - window_size)
+    return mean(history[min_index:end])
+end
+
+function estimate_energy_with_samples(prob, samples_stop, samples_logprob, samples_logprob_stop, baseline)
+	fv = Base.Fix1(LogDensityProblems.logdensity, prob).(eachsample(samples_stop))
+	fv_mean = mean(fv) 
+	score_grad = mean(samples_logprob .* (fv .- baseline))
+	score_grad_stop = mean(samples_logprob_stop .* (fv .- baseline))
+	return fv_mean + (score_grad - score_grad_stop)
+end
+
+function compute_elbo(q, q_stop, samples_stop, entropy, problem, baseline)
+	samples_logprob = logpdf.(Ref(q), AdvancedVI.eachsample(samples_stop))
+	samples_logprob_stop = logpdf.(Ref(q_stop), AdvancedVI.eachsample(samples_stop))
+	energy = estimate_energy_with_samples(problem, samples_stop, samples_logprob, samples_logprob_stop, baseline)
+	elbo = energy + entropy
+	return elbo
+end
+
+function estimate_objective(
+	rng::Random.AbstractRNG,
+	obj::ScoreGradELBO,
+	q,
+	prob;
+	n_samples::Int = obj.n_samples,
+)
+	samples, entropy = reparam_with_entropy(rng, q, q, n_samples, obj.entropy)
+	energy = Base.Fix1(LogDensityProblems.logdensity, prob).(eachsample(samples))
```

Comment:
```suggestion
	energy = map(Base.Fix1(LogDensityProblems.logdensity, prob), eachsample(samples))
```
I feel that the `.` operator here is a little too sneaky. :wink: